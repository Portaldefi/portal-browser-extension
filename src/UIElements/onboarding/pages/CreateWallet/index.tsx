import React, { useEffect, useCallback } from 'react';
import { useNavigate } from 'react-router-dom';
import { Form, Grid, Header, Segment } from 'semantic-ui-react';

import { setItems } from '../../slices/menuSlice';
import { setSRFLength, setSRFList } from '../../slices/phraseSlice';
import { useAppDispatch } from '../../hooks';

export default () => {
  const dispatch = useAppDispatch();
  const navigate = useNavigate();

  useEffect(() => {
    dispatch(setItems([
      {
        title: 'Back',
        link: '/select-action',
      }
    ]));
  }, []);

  const handleCreate = useCallback(() => {
    // TODO: Backend interaction
    dispatch(setSRFLength(12));
    dispatch(setSRFList([
      'minor', 'regret', 'daring', 'perfect', 'tenant', 'surge', 'they', 'section', 'mobile', 'bottom', 'vacant', 'cheese'
    ]));

    navigate('/seed-phrase-intro');
  }, []);
  
  return (
    <Segment size='big'>
      <Header size='large' as='h1'>Create Password</Header>
      <Grid centered columns={1}>
        <Grid.Column width='eight'>
          <Form onSubmit={handleCreate}>
            <Form.Field>
              <Form.Input label='New password (8 characters min)' size='big' type='password' />
            </Form.Field>
            <Form.Field>
              <Form.Input label='Confirm password' size='big' type='password' />
            </Form.Field>
            <Form.Field>
              <Form.Checkbox label={{children: <div>I have read and agree to the <a>Terms of Use</a></div>}} />
            </Form.Field>
            <Form.Button type='submit' size='large' color='blue'>
              Create
            </Form.Button>
          </Form>
        </Grid.Column>
      </Grid>
    </Segment>
  )
}