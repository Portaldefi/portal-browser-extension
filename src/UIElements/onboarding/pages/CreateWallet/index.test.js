import CreateWallet from './index';
import { create, act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store'
import { Button, Form } from 'semantic-ui-react';

describe('CreateWalletComponent', () => {
  let initialState;
  let mockStore;
  let store;
  let jsdomAlert;
  beforeEach(() => {
    initialState = {
      phrase: {
        SRF_Length: 0,
        SRF_List: [],
        password: ''
      },
      menu: {
        items: []
      }
    }
    mockStore = configureStore()
    store = mockStore(initialState);
    jsdomAlert = window.alert;  // remember the jsdom alert
    window.alert = () => {};  // provide an empty implementation for window.alert
  });

  it('should work', async () => {
    let tree;
    act(() => {
      tree = create(
        <Provider store={store}>
          <BrowserRouter>
            <CreateWallet />
          </BrowserRouter>
        </Provider>
      );
    });

    expect(tree).toMatchSnapshot();
  });

  it('functions', async () => {
    let tree;
    act(() => {
      tree = create(
        <Provider store={store}>
          <BrowserRouter>
            <CreateWallet />
          </BrowserRouter>
        </Provider>
      );
    });

    const buttons = tree.root.findAllByType(Button);
    act(() => buttons[0].props.onClick());
    expect(tree).toMatchSnapshot();
    act(() => buttons[1].props.onClick());
    expect(tree).toMatchSnapshot();

    const inputs = tree.root.findAllByType(Form.Input);
    act(() => inputs[0].props.onChange({
      target: {
        value: "1123"
      }
    }));
    expect(tree).toMatchSnapshot();
  });
  afterAll(() => {
    jest.resetModules();
    window.alert = jsdomAlert;  // restore the jsdom alert
  });
});
