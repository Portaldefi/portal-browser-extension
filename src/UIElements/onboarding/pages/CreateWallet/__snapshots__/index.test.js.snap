// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateWalletComponent should work 1`] = `
<div
  className="ui segment board"
>
  <div
    className="ui grid"
  >
    <div
      className="centered row"
    >
      <div
        className="ui medium header heading"
      >
        Create Password
      </div>
    </div>
    <div
      className="centered row"
    >
      <form
        className="ui form form-container"
        onSubmit={[Function]}
      >
        <div
          className="inline one fields"
        >
          <div
            className="eight wide field"
          >
            <label>
              Password
            </label>
            <div
              className="ui input"
            >
              <input
                aria-describedby={null}
                name="password"
                onChange={[Function]}
                type="password"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="inline one fields"
        >
          <div
            className="eight wide field"
          >
            <label>
              Confirm
            </label>
            <div
              className="ui input"
            >
              <input
                aria-describedby={null}
                name="passConfirm"
                onChange={[Function]}
                type="password"
                value=""
              />
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      className="row"
    >
      <p
        className="ui grey small header description extra-former-blank extra-latter-blank"
      >
        WARNING: Improper backup of the seed phrase will result in loss of funds!
      </p>
    </div>
    <div
      className="centered two column row"
    >
      <div
        className="column"
      >
        <button
          className="ui button blank-button"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="angle left icon"
            onClick={[Function]}
          />
          Back
        </button>
      </div>
      <div
        className="column"
      >
        <button
          className="ui button primary-button"
          onClick={[Function]}
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</div>
`;
